skinfer/schema_inferer.py

```diff
@@ -17,6 +18,8 @@ def load_samples_from_jsonlines(file_list):
     for filename in file_list:
         with gzopen(filename) as f:
             for line in f:
+                if isinstance(line, six.binary_type):
```

Comment:
six.binary_type is for backwards compatibility with python 2.5, you can use `bytes`.

I wonder if it is possible to eliminate this type check though - why should `gzopen` open files both in binary and text modes?